README.md

```diff
@@ -1,41 +1,74 @@
-When you add a checkpoint, a metadata file describing the contents of the
-checkpoint (shape, dtype and hash for each parameter group)
-is staged by `git` (and will be what is pushed to a remote),
-but the local copy of your checkpoint remains unchanged.
-During a checkout operation, the checkpoint is reconstituted locally based
-on the contents of `.git_cml/{checkpoint_file_name}`.
-
-For a given parameter group, `git-cml` will store either the group's parameter's
-values or an update to the values in the event that it can be stored more
-efficiently.
-For example, if a sparse update is made to a parameter group, `git-cml` will
-store the sparse update in a sparse format to save storage costs.
-For a given parameter group, all updates made since the last time the group's
-values were stored are stored in an `updates` subdirectory within
-`.git_cml/{checkpoint_file_name}/{parameter_name}`.
-Whenever a dense update is made to a parameter group, the new full set of
-parameter values is stored and all prior updates are removed from the
-`updates` subdirectory.
-All parameters and updates are currently stored via `git-lfs`.
-
-`git diff` will identify which parameter groups have changed and how.
-
-`git merge` will assume that all merges to the checkpoint (i.e. to parameter
-group files) result in merge conflicts and offer various possible automated
-merging strategies that can be tried and vetted.
+# git-theta
+
+git extension for collaborative, continual, and communal model development. 
+
+# How to use this repository?
+## LFS installation 
+Download the LFS package from the [website](https://github.com/git-lfs/git-lfs/releases/tag/v3.2.0). For Linux users, download the amd64 version from the list of assests in the website. 
+
+## Getting started
+clone the repository  
+```bash
+git clone https://github.com/r-three/git-theta.git
+```
+Install the packages by running:
+```bash
+cd git-theta
+pip install -e .
+```
+## Installing git theta
+<!--Is repository same as codebase?-->
+You can initialize the git theta in the home directory of the codebase to track code and models as follows:
+```bash
+git theta install
+```
+
+The following lines will be added to the `.gitconfig` file in the root directory of the user after the successful installation. 
+```
+[filter "lfs"]
+        smudge = git-lfs smudge -- %f
+        required = true
+        clean = git-lfs clean -- %f
+[filter "theta"]
+        clean = git-theta-filter clean %f
+        smudge = git-theta-filter smudge %f
+        required = true
+```
+
+# Example Usage
+<!--Create a folder with a text file and a model checkpoint. Initialize it as a git repository.-->
+First, initialize git in the home directory of the codebase
+```bash
+git init
+```
+In order to track the model checkpoint using git theta, run this command
+```bash 
+git theta track {path_to_model_checkpoint}
+```
+
+The above command adds the following lines to the `.gitattributes` files in the home directory.
```

Comment:
Probably more of a design question than something to address now, but there seems to be some inconsistency on if things are done at a per-repo level or globally. Some of the git commands have `--global` flags we might want to copy.

Reply:
This is actually a mistake in the README. The `.gitattributes` file that gets written is actually in the root of the git repository, not in the user's home directory. 

As for why some configuration is at the user-level in `~/.gitconfig` and some is at the repo-level in `$git_root/.gitattributes`, this follows the git-lfs design. My understanding of why they chose this design is as follows:

Upon `git clone`, the filters needed for checking out model checkpoint files must already be present.  That is why the user must run `git theta install` to define these filters in the user-level `~/.gitconfig` before even cloning a git-theta repo. However, the paths to the checkpoint files in the repo are repo-specific and so they must be stored in the repo-level `$git_root/.gitattributes` file and must actually be version controlled.

Reply:
Interesting, basically a bootstrapping problem. Makes sense.

We might want to document that it is in-line with `git lfs` (especially so you don't have to type this out again when I forget and ask why we are going this again lol).